packages/roc-package-web-app-react/app/server/render.js

```diff
@@ -99,13 +96,11 @@ export function reactRender(url, createRoutes, store, renderPage, staticRender =
                     };
                 }
 
-                getPrefetchedData(components, locals)
+                trigger('fetch', components, locals)
                     .then(() => {
-                        let component = <RoutingContext {...renderProps} />;
+                        let component = <RouterContext {...renderProps} />;
```

Comment:
Any chance of, instead of triggering `'fetch'`, taking in a customizable function that can trigger whatever it needs. The default implementation, if not overridden, runs this triggering logic. If we go that route, we allow end users to hook into the server side logic and define additional fetching methods, such as optional data fetches that are allowed to fail and also triggering the deferred data fetches on the server. In this way, we can defer data fetching on the client, allowing the router to render the new route, while still allowing the server to fetch "all" data regardless of whether it happens before or after router transition on the client